### PR TITLE
Make sure library code is completely separated from cli code

### PR DESCRIPTION
**Pull request by matthewmcgarvey**
Fixes https://github.com/luckyframework/lucky_cli/issues/620

This is a pretty big change, what's the point?

Well as #620 points out, the [teeplate shard](https://github.com/luckyframework/teeplate/blob/797fc0c5414d8eff06b6d76c04d9be881623a636/src/lib/file_tree.cr#L7-L9) has some code that causes compilation to slow down by 10 seconds when it can't use the cache. All that for code that doesn't actually need to run except for when the shard is built to be run as a cli.

Almost all of the code movement is because I want to make sure we never accidentally include the cli code in the library code going forward.

### CODE REVIEW

**Comment by matthewmcgarvey** [comment]
Before approving/merging this, let's consider pulling the task code into a separate shard so that we avoid circular dependencies

File: src/cli/templates/browser_authentication_src_template (Copy 1).cr [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class BrowserAuthenticationSrcTemplate < Teeplate::FileTree
```
Comment (jwoertink) [inline src/cli/templates/browser_authentication_src_template (Copy 1).cr:1]: The filename on this says "Copy 1"

**Comment by jwoertink** [comment]
wow. intense 😂  The end result will be worth it though. Since this is sort of a big thing, let me know what you'd like me to do. I don't want to just hop in and step on your toes. If you want me to look at breaking out the tasks deal, I can start there. Or unless you've already started on that..

**Comment by matthewmcgarvey** [comment]
Closing in favor of https://github.com/luckyframework/lucky_cli/pull/622

**Pull request closed without merging**